### PR TITLE
Modernize "Hello World" page of the manual a bit

### PR DESCRIPTION
**Pull request by vadz**
The main change is the use of `Bind()` instead of the event table. I also wonder if we shouldn't give a C++11 example here, as using `Bind()` with lambdas is really convenient.

Any other suggestions for enhancements?

### CODE REVIEW

**Comment by pbfordev** [comment]
I guess demonstrating Bind() with a lambda would be nice, provided the added C++11 check doesn't clutter the example much.

Perhaps adding wxOVERRIDE to MyApp::OnInit() declaration would not hurt either.

File: docs/doxygen/overviews/helloworld.h [hunk 1]
```diff
@@ -83,59 +84,46 @@ enum
...
+    MyFrame *frame = new MyFrame();
     frame->Show( true );
     return true;
 }
 @endcode
 
 In the constructor of the main window (or later on) we create a menu with our
 menu items as well as a status bar to be shown at the bottom of the main
-window. Both have to be associated with the frame with respective calls.
+window. Both have to be associated with the frame with respective calls. We
```
Comment (iwbnwif) [inline docs/doxygen/overviews/helloworld.h]: Suggest to start "We ..." as a new paragraph to help readability when adding the text I suggest below.

File: docs/doxygen/overviews/helloworld.h [hunk 2]
```diff
@@ -83,59 +84,46 @@ enum
...
+also need to actually connect our event handlers to the events we want to
+handle in them, by calling Bind() to send all the menu events with the
```
Comment (iwbnwif) [inline docs/doxygen/overviews/helloworld.h]: ... all the menu events (identified by wxEVT_MENU) ...

File: docs/doxygen/overviews/helloworld.h [hunk 3]
```diff
@@ -83,59 +84,46 @@ enum
...
+specified ID to the given function:
```
Comment (iwbnwif) [inline docs/doxygen/overviews/helloworld.h]: "... function: " change to ... function. The 4 parameters for this version of Bind() are: the event type; the function or functor to be called; the window whose event queue we are binding to; and the user or system identifier that differentiates between events of the same type.

**Comment by iwbnwif** [comment]
I added some comments against lines in the file. The reason for adding more explicit detail about Bind() is that IMHO almost all the other code is pretty straightforward C++ / GUI library stuff, but Bind() is rather specific to wxWidgets.

**Comment by vadz** [comment]
@pbfordev I think speaking about `wxOVERRIDE` here would be distracting, it's not really wx-specific. People using C++11 could/should just use `override` directly.

@iwbnwif Thanks for your comments, I've tried to address them in the new version.

File: docs/doxygen/overviews/helloworld.h [hunk 4]
```diff
@@ -163,6 +149,32 @@ translated) labels and also standard accelerators correct for the current
...
 
+We also have to actually connect our event handlers to the events we want to
+handle in them, by calling Bind() to send all the menu events, identified by
+wxEVT_MENU event type, with the specified ID to the given function. The
+parameters we pass to Bind() are
+
+-# The event type, e.g. wxEVT_MENU or wxEVT_BUTTON or wxEVT_SIZE or one
+   of many other events used by wxWidgets.
+-# Pointer to the member function to call and the object to call it on. In
+   this case we just call our own function, but we could call a member
```
Comment (iwbnwif) [inline docs/doxygen/overviews/helloworld.h]: ... call our own function (hence 'this' for the object), but we could call ... maybe it is too obvious, but personally I think it's worth being clear why there is a 'this' in the call.

**Comment by iwbnwif** [comment]
Thanks, even better! Just one further suggestion (ignore if you want)

**Pull request closed without merging**